skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,316 @@
+
+    Attributes
+    ----------
+    spline_mappings : (N, D) array_like
+        Spline mappings for every control point.
+    src : (N, 2) array_like
+        Coordinates of control points in source image.
+
+    References
+    ----------
+    .. [1] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+           decomposition of deformations," IEEE Transactions on pattern analysis
+           and machine intelligence 11.6 (1989): 567–585. DOI:`10.1109/34.24792`
+
+    Examples
+    --------
+    >>> import skimage as ski
+
+    Define source and destination control points:
+
+    >>> src = np.array([[0, 0], [0, 5], [5, 5],[5, 0]])
+    >>> dst = np.roll(src, 1, axis=0)
+
+     Generate meshgrid:
+
+    >>> coords = np.meshgrid(np.arange(5), np.arange(5))
+    >>> t_coords = np.vstack([coords[0].ravel(), coords[1].ravel()]).T
+
+     Estimate transformation:
+
+    >>> tps = ski.transform.TpsTransform()
+    >>> tps.estimate(src, dst)
+    True
+
+    Apply the transformation:
+
+    >>> trans_coord = tps(t_coords)
+    >>> xx_trans = trans_coord[:, 0]
+    >>> yy_trans = trans_coord[:, 1]
+    >>> coords[1]
+    array([[0, 0, 0, 0, 0],
+           [1, 1, 1, 1, 1],
+           [2, 2, 2, 2, 2],
+           [3, 3, 3, 3, 3],
+           [4, 4, 4, 4, 4]])
+    >>> expected_yy = np.array([0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4,
+    ...                         0, 1, 2, 3, 4])
+    >>> np.allclose(yy_trans, expected_yy)
+    True
+    """
+
+    def __init__(self):
+        self._estimated = False
+        self.spline_mappings = None
+        self.src = None
+
+    def __call__(self, coords):
+        """Estimate the transformation from a set of corresponding points.
+
+        Parameters
+        ----------
+        coords : (N, 2) array_like
+            x, y coordinates to transform
+
+        Returns
+        -------
+        transformed_coords: (N, D) array
+            Destination coordinates
+        """
+        if self.spline_mappings is None:
+            raise ValueError(f"{self.spline_mappings}. Compute the `estimate`")
+        coeffs = self.spline_mappings
+        coords = np.array(coords)
+
+        if not coords.ndim == 2 or coords.shape[1] != 2:
+            raise ValueError("Input `coords` must have shape (N, 2)")
+
+        radial_dist = self._radial_distance(coords[:, 0], coords[:, 1])
+
+        x_warp = self._spline_function(
+            coords[:, 0], coords[:, 1], radial_dist, coeffs[:, 0]
+        )
+        y_warp = self._spline_function(
+            coords[:, 0], coords[:, 1], radial_dist, coeffs[:, 1]
+        )
+        return np.vstack([x_warp, y_warp]).T
+
+    @property
+    def inverse(self):
+        raise NotImplementedError("This is yet to be implemented.")
+
+    def estimate(self, src, dst):
+        """Estimate optimal spline mappings that describe the deformation of the points.
```

Comment:
```suggestion
        """Estimate optimal spline mappings that describe the deformation of source
        points into destination points.
```
Is the final -s in 'mappings' intended? Does the function estimate several mappings (in general)?